src/index.ts

```diff
@@ -21,44 +25,85 @@ async function run(): Promise<void> {
   // Expected description format - "2019-07-18.1563481767-1-g7f60" < 'g<first 4 sha char>'
   //       commit date branched from ^     same sec ^     ^ commit number on branch (0 is the commit you branched from)
   setVariable(EnvironmentVariables.CI_COMMIT_DESCRIPTION, '');
 
   const nowMilliseconds = Date.now();
-  const nowSeconds = Math.floor(nowMilliseconds/1000);
+  const nowSeconds = Math.floor(nowMilliseconds / 1000);
   const nowAsTimeIsoString = new Date(nowMilliseconds).toISOString();
-  const nowAsTimeExpectedIsoString = nowAsTimeIsoString.replace(/\.\d{3}Z/,'Z'); // No millis expect
+  const nowAsTimeExpectedIsoString = nowAsTimeIsoString.replace(
+    /\.\d{3}Z/,
+    'Z',
+  ); // No millis expect
   setVariable(EnvironmentVariables.CI_TIMESTAMP, nowSeconds.toString());
   setVariable(EnvironmentVariables.CI_STRING_TIME, nowAsTimeExpectedIsoString);
 
-
   const branchName = context?.ref?.match(/[^/]+$/)?.[0];
-  if(branchName) {
+  if (branchName) {
     setVariable(EnvironmentVariables.CI_BRANCH, branchName);
   }
 
-  // @ts-ignore - missing lib defs
-  setVariable(EnvironmentVariables.CI_REPO_NAME,  event?.repository?.name );
+  setVariable(EnvironmentVariables.CI_REPO_NAME, event?.repository?.name);
+}
 
-  // TODO The pull request payload is missing in this implementation (tested with PR builds)
-  // Note: the PR functionality in Codeship is not function from my empirical testing, so this behavior is the same as Codeship
-  const pullRequest = event?.pull_request;
-  setVariable(EnvironmentVariables.CI_PULL_REQUEST,  pullRequest?.number?.toString() ?? '');
-  setVariable(EnvironmentVariables.CI_PR_NUMBER,  pullRequest?.html_url ?? 'false');
+function populatePushEventCommitDetails(
+  pushEvent: PushEvent,
+  context: Context,
+) {
+  const headCommit = pushEvent.head_commit;
+  const author = headCommit?.author;
+  setVariable(EnvironmentVariables.CI_COMMIT_ID, context?.sha);
+  setVariable(EnvironmentVariables.CI_COMMIT_MESSAGE, headCommit?.message);
+  setVariable(EnvironmentVariables.CI_COMMITTER_USERNAME, author?.username);
+  setVariable(EnvironmentVariables.CI_COMMITTER_EMAIL, author?.email ?? '');
+  setVariable(EnvironmentVariables.CI_COMMITTER_NAME, author?.name);
+}
+
+async function populatePullRequestEventCommitDetails(
+  pullRequestEvent: PullRequest,
+  context: Context,
+) {
+  const head = pullRequestEvent.head;
+  const user = head.user;
+  setVariable(EnvironmentVariables.CI_COMMIT_ID, head.sha);
+  setVariable(EnvironmentVariables.CI_COMMITTER_USERNAME, user.login);
+  setVariable(EnvironmentVariables.CI_COMMITTER_EMAIL, user.email ?? '');
+  setVariable(EnvironmentVariables.CI_COMMITTER_NAME, user.name);
+  setVariable(EnvironmentVariables.CI_PULL_REQUEST, pullRequestEvent.url);
+  setVariable(
+    EnvironmentVariables.CI_PR_NUMBER,
+    pullRequestEvent.number?.toString(),
+  );
+
+  const GITHUB_TOKEN = core.getInput('github-token');
+  if (GITHUB_TOKEN) {
```

Comment:
Should we issue a logging WARN or otherwise indicate that this is a PR action, but we don't have the token, so we're not able to lookup the actual commit message?